### DISCUSSION TITLE
Inventory Question

### DISCUSSION THREAD

**Discussion by Nopock** [Minestom/Minestom#1278] [Help]
`
abstract class Menu(val type: InventoryType) {

    abstract fun getTitle(player: Player): String

    abstract fun getButtons(player: Player): Map<Int, Button>

    fun openMenu(player: Player) {
        player.closeInventory()

        val inventory = Inventory(type, translate(getTitle(player)))

        val buttons = getButtons(player)

        buttons.forEach { (slot, button) ->
            inventory.setItemStack(slot, button.getItem(player))
        }

        inventory.addInventoryCondition { p: Player, slot: Int, type: ClickType, inventoryConditionResult: InventoryConditionResult ->
            if (buttons.containsKey(slot)) {
                buttons[slot]?.clicked(p, slot, type, inventoryConditionResult)
            }
        }

        player.openInventory(inventory)
    }
}`

Doesn't seem to add items to the menus

`
    override fun getButtons(player: Player): Map<Int, Button> {
        val buttons = mutableMapOf<Int, Button>()

        buttons[10] = getButton(Punishment.Type.BLACKLIST)
        buttons[12] = getButton(Punishment.Type.BAN)
        buttons[14] = getButton(Punishment.Type.MUTE)
        buttons[16] = getButton(Punishment.Type.WARN)

        return buttons
    }
`

**Comment by GreatWyrm** [comment]
There's not enough information to properly diagnose the issue, but let me offer up a few areas you can look in.

1. Are there any errors logged in console? Any exceptions/stacktraces?
2. Double check that you are running this code, maybe add a print statement at the end of this code to make sure you are running it.
3. Is the inventory type large enough to hold the buttons? I'd assume so, but I'd double check that real quick
4. You may want to look into the `button.getItem(player)` function, see if that is returning the proper ItemStack.

But if the code is working fine, you will have to figure out what's going wrong with your `Inventory.setItemStack(Slot, ItemStack)` function calls.
